File: server/app/Genre.php
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Genre extends Model
{
    protected $table      = 'genre';
    protected $primaryKey = 'slug';

    public $timestamps   = false;
    public $incrementing = false;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'slug',
        'name'
    ];

    protected $appends = [
        'image_link'
    ];

    public function news_outlet_genre() {
        return $this->hasMany(NewsOutletGenre::class, 'genre_slug', 'slug');
    }

    public function getImageLinkAttribute() {
        return app()->basePath('public/images/genres/' . $this->attributes['slug'] . '.jpg');
    }
}

File: android/app/src/main/java/uk/co/tobymellor/compress/models/genres/Genre.java
package uk.co.tobymellor.compress.models.genres;

import uk.co.tobymellor.compress.models.Model;

public class Genre extends Model {
    private final String slug;
    private final String name;
    private final String imageLink;

    public Genre(GenreInput input) {
        this.slug      = input.getSlug();
        this.name      = input.getName();
        this.imageLink = input.getImageLink();
    }

    public String getSlug() {
        return slug;
    }

    public String getName() {
        return name;
    }

    public String getImageLink() {
        return imageLink;
    }
}
